specs/altair/sync-protocol.md

```diff
@@ -101,50 +102,80 @@ def get_subtree_index(generalized_index: GeneralizedIndex) -> uint64:
     return uint64(generalized_index % 2**(floorlog2(generalized_index)))
 ```
 
+### `get_active_header`
+
+```python
+def get_active_header(update: LightClientUpdate) -> BeaconBlockHeader:
+    # Is the update trying to convince us of a finalized header or an optimistic header?
+    if update.finalized_header != BeaconBlockHeader():
+        return update.finalized_header
+    else:
+        return update.attested_header
+```
+
+### `get_safety_threshold`
+
+```python
+def get_safety_threshold(store: LightClientStore):
+    return max(
+        store.previous_period_max_attendance,     
+        store.current_period_max_attendance
+    ) // 2
+```
+
 ## Light client state updates
 
-A light client maintains its state in a `store` object of type `LightClientStore` and receives `update` objects of type `LightClientUpdate`. Every `update` triggers `process_light_client_update(store, update, current_slot)` where `current_slot` is the current slot based on some local clock.
+A light client maintains its state in a `store` object of type `LightClientStore` and receives `update` objects of type `LightClientUpdate`. Every `update` triggers `process_light_client_update(store, update, current_slot)` where `current_slot` is the current slot based on some local clock. `process_slot` is processed every time the current slot increments.
+
+### `process_slot`
+
+```python
+def process_slot(store: LightClientStore, current_slot: Slot):
+    if current_slot % SAFETY_THRESHOLD_CALCULATION_PERIOD == 0:
+        store.previous_period_max_attendance = store.current_period_max_attendance
+        store.current_period_max_attendance = 0
+```
 
 #### `validate_light_client_update`
 
 ```python
-def validate_light_client_update(snapshot: LightClientSnapshot,
+def validate_light_client_update(store: LightClientStore,
                                  update: LightClientUpdate,
                                  genesis_validators_root: Root) -> None:
-    # Verify update slot is larger than snapshot slot
-    assert update.header.slot > snapshot.header.slot
+                                 
+    # Verify update slot is larger than slot of current best finalized header
+    active_header = get_active_header(update)
+    assert active_header.slot > store.finalized_header.slot
```

Comment:
What if `active_header.slot` happens to be in the future of the caller's `current_slot`?